nilearn/signal.py

```diff
@@ -732,8 +738,12 @@ def clean(
 
     # Interpolation / censoring
     signals, confounds = _handle_scrubbed_volumes(
-        signals, confounds, sample_mask, filter_type, t_r
+        signals, confounds, sample_mask, filter_type, t_r, extrapolate
     )
+    if sample_mask is not None and not extrapolate:
+        # reset the indexing of the sample_mask excluding non-interpolated
+        # volumes at the head of the data
+        sample_mask -= sample_mask[0]
```

Comment:
Sorry @htwangtw but I am afraid I can't follow you fully. If you refer to the data type (list vs numpy.array), then indeed it won't work unless you cast your list of indexes.  But not because of any of the changes introduced, simply because sanitizing `_sanitize_sample_mask()` function will raise a TypeError.  
Otherwise, if you refer to the fact of having a sample_mask which labels your n first volumes as valid/good, the index resetting operation stills valid. Nothing should be modified as is the case when interpolation gets rid of the volumes at the end of the data run, solely when volumes are at the beginning of the run, you need to reset the `sample_mask` indexes (unless I am missing something).

Reply:
Ah thanks for the elaboration! Now this is clearer to me